WooCommerce/Classes/ViewRelated/JetpackSetup/JetpackSetupCoordinator.swift

```diff
@@ -102,12 +105,15 @@ private extension JetpackSetupCoordinator {
     }
 
     /// Checks the Jetpack connection status for non-Jetpack sites to infer the setup steps to be handled.
-    func checkJetpackStatus(_ result: Result<JetpackUser, Error>, skipsWPComLogin: Bool = false) {
+    func checkJetpackStatus(_ result: Result<JetpackUser, Error>, onSuccess: (() -> Void)? = nil) {
```

Comment:
Can we add comments to explain when `onSuccess` will be called and what happens if we don't provide it? 

I see that we call `startAuthentication` and `checkAdminRoleAndStartLoginIfPossible` if `onSuccess` is not provided. Do we have to provide `onSuccess` to skip/avoid authentication and return early?

Reply:
This logic looks pretty messy so I updated it in 62be3ed72f2e3118a46cb102795f867af6669d37. The method is now named `retrieveJetpackConnectedEmail`, which returns the connected email if available or throw error otherwise.